### PR TITLE
Docs: Add syntax for registering backends in `pyproject.toml`

### PR DESCRIPTION
**Pull request by ZedThree**
Uses `sphinx-inline-tabs` for pretty and compact options.

Also reword surrounding text a little for clarity.

### CODE REVIEW

**Comment by keewis** [comment]
this does look like a good addition to me, thanks.

Could you run the `pre-commit` hooks, though? It seems you created the PR from a branch on a organization, so the bot can't do that for you.

**Comment by ZedThree** [comment]
Done! :+1:

File: doc/internals/how-to-add-new-backend.rst [hunk 1]
```diff
@@ -267,42 +268,57 @@ interface only the boolean keywords related to the supported decoders.
...
 
-    setuptools.setup(
-        entry_points={
-            "xarray.backends": ["my_engine=my_package.my_module:MyBackendEntryClass"],
-        },
-    )
+   .. code:: toml
+
+      [project.entry-points."xarray-backends"]
+      my_engine = "my_package.my_module:MyBackendEntryClass"
```
Comment (ZedThree) [inline doc/internals/how-to-add-new-backend.rst:285]: Actually maybe this should be 

```suggestion
      my_engine = "my_package.my_module:MyBackendEntrypoint"
```

to match [the example](https://docs.xarray.dev/en/latest/internals/how-to-add-new-backend.html#backendentrypoint-subclassing)?

Comment (dcherian) [inline doc/internals/how-to-add-new-backend.rst:285]:
```suggestion
      my_engine = "my_package.my_module:MyBackendEntrypoint"
```

Comment (dcherian) [reply doc/internals/how-to-add-new-backend.rst:285]: Yes lets do that. (I can't apply that change for some reason)_

Comment (keewis) [reply doc/internals/how-to-add-new-backend.rst:285]: I think that's also caused by access restrictions to the repositories in organizations (not sure though).

Comment (ZedThree) [reply doc/internals/how-to-add-new-backend.rst]: Hmm, thought I had allowed maintainers to modify the branch -- no worries, I've made the change! :)

**Comment by dcherian** [comment]
Perfect. Thank you!

**Pull request merged**